File: 42trappingRainWater/trappingRainWater.py
class Solution:
    def trap(self, height: List[int]) -> int:
        result = level = 0
        left, right = 0, len(height) - 1
        while left < right:
            low = min(map(height.__getitem__, (left, right)))
            (left:= left + 1) if operator.lt(*map(height.__getitem__, (left, right))) else (right := right - 1)
            level = max(level, low)
            result += level - low
        return result

File: 42trappingRainWater/trappingRainWater.java
class Solution {
    public int trap(int[] height) {
        int result = 0, level = 0;
        int left = 0, right = height.length - 1;
        while (left < right)
        {
            final int low = java.util.stream.IntStream.of(left, right).map($ -> height[$]).min().orElse(0);
            int $_ = java.util.stream.IntStream.of(left, right).map($ -> height[$]).reduce((a ,b) -> a - b).orElse(0) < 0 ? ++left : --right;
            level = Math.max(level, low);
            result += level - low;
        }
        return result;
    }
}
